client/session.go

```diff
@@ -170,8 +192,38 @@ func (s *session) setTopologyMap(topologyMap m3db.TopologyMap) error {
 	s.Lock()
 	prev := s.queues
 	s.queues = queues
-	s.setWriteOpPoolsWithLock(topologyMap)
 	s.topoMap = topologyMap
+	prevReplicas := s.replicas
+	s.replicas = topologyMap.Replicas()
+	s.quorum = topologyMap.QuorumReplicas()
+	if s.fetchBatchOpArrayArrayPool == nil ||
+		s.fetchBatchOpArrayArrayPool.Entries() != len(queues) {
+		s.fetchBatchOpArrayArrayPool = newFetchBatchOpArrayArrayPool(
+			s.opts.GetFetchBatchOpPoolSize(),
```

Comment:
Do we need a separate size for this pool? Feels like the default fetchBatchOpPoolSize should be bigger than the default fetchBatchOpArrayArrayPoolSize, no?